src/main/java/io/tabular/AsyncFileIO.java

```diff
@@ -412,6 +415,7 @@ public long getLength() {
       if (future.isDone()) {
         return delegate().getLength();
       } else {
+        future.cancel(true);
```

Comment:
Probably we should check the return value and log if it's false. That indicates that most likely the task had *just* completed but we still have to do the remote read. Or better yet, if it's false, do the disk read since it's there